pysumma/simulation.py

```diff
@@ -41,190 +40,74 @@ def __init__(self, executable, filemanager, run_suffix='_pysumma_run'):
+                  progress=None):
+        run_cmd = self.gen_summa_cmd(processes, prerun_cmds,
+                                     startGRU, countGRU, iHRU, freq_restart,
+                                     progress)
+        self.process = subprocess.Popen(run_cmd, stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE, shell=True)
+        self._status = 'Running'
 
+    def run_docker(self, docker_img, run_suffix=None, processes=1,
+                   prerun_cmds=[], startGRU=None, countGRU=None, iHRU=None,
+                   freq_restart=None, progress=None):
         self.executable = docker_img
-        in_path = self.manager.input_path.value
-        out_path = self.manager.output_path.value
-        self.file_dir = os.path.dirname(self.manager_path)
-        base_cmd = "docker run -v {}:{}".format(
-            self.file_dir, self.file_dir)
-        if self.file_dir+'/' == self.manager.settings_path.value:
-            cmd = (base_cmd
-                   + " -v {}:{}".format(in_path, in_path)
-                   + " -v {}:{}".format(out_path, out_path)
-                   + " {} -p never -s {} -m {}".format(self.executable,
-                                                       run_suffix,
-                                                       self.manager_path))
-        else:
-            setting_path = self.manager.settings_path.value
-            cmd = (base_cmd
-                   + " -v {}:{}".format(setting_path, setting_path)
-                   + " -v {}:{}".format(in_path, in_path)
-                   + " -v {}:{}".format(out_path, out_path)
-                   + " {} -p never -s {} -m {}".format(self.executable,
-                                                       run_suffix,
-                                                       self.manager_path))
-        # run shell script in python and print output
-        cmd = shlex.split(cmd)
-        p = subprocess.Popen(cmd, stdout=subprocess.PIPE)
-        output = p.communicate()[0].decode('utf-8')
-        print(output)
-        if 'FATAL ERROR' in output:
-            raise Exception("SUMMA failed to execute!")
-        # define output file name as sopron version of summa
-        if self.manager.output_path.value.split('/')[0] == '<BASEDIR>':
-            out_file_path = (
-                self.manager.output_path.value.split('<BASEDIR>')[1]
-                + self.manager.output_prefix.value + '_output_'
-                + run_suffix + '_timestep.nc')
-        else:
-            out_file_path = (
-                self.manager.output_path.value
-                + self.manager.output_prefix.value + '_output_'
-                + run_suffix + '_timestep.nc')
-        return out_file_path
+        run_cmd = self.gen_summa_cmd(processes, prerun_cmds,
+                                     startGRU, countGRU, iHRU,
+                                     freq_restart, progress)
+
+        fman_dir = os.path.dirname(self.manager_path.value)
```

Comment:
self.manager_path.value -> self.manager_path
There is no value in manager_path.

Reply:
Thanks, will fix this.